### DISCUSSION TITLE
Force a different dependency version

### DISCUSSION THREAD

**Discussion by hacker-DOM** [Homebrew/discussions#4024] [Everyday usage] [answered]
I'd like to install pipx with homebrew. However, pipx currently lists python@3.11 as a dependency. The application I want to install with pipx does not support python 3.11.

(I've tried installing python@3.10 and then pipx, but it tries to install python@3.11)

Is it possible to force the pipx formula to use python@3.10?

**Comment by SMillerDev** [comment, marked as answer]
No, you can't mix and match dependencies in Homebrew

**Comment by carlocab** [comment]
You can create your own version of the `pipx` formula in your own tap using `brew tap-new` and `brew extract`. See `man brew` for details.

Here are some other docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Comment by alebcay** [comment]
Another option is to avoid installing `pipx` via Homebrew. Instead, install `python@3.10` and install `pipx` via `python@3.10`'s `pip`:
```
brew install python@3.10
python3.10 -m pip install pipx
pipx [...]
```

You can also consider installing tools like `asdf` or `pyenv` with Homebrew and let `asdf` or `pyenv` manage Python versions for you, which you can then use to install `pipx` in via `pip`. See [this blog post](https://justinmayer.com/posts/homebrew-python-is-not-for-you/) for more about this idea/approach/rationale.

**Comment by martina-oefelein** [comment]
`pipx` can install applications using a different Python version than its own. So you can use the brew-provided pipx (with Python 3.11) to install an application that requires Python 3.10:

```bash
brew install pipx    # Install pipx using Python 3.11
brew install python@3.10   # Install Python 3.10 for your-app
pipx install --python python3.10 your-app    # Install your-app using Python 3.10
```
